pkg/apis/authentication/types_credentialsbinding.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package authentication
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CredentialsBinding represents a binding to credentials in the same or another namespace.
+type CredentialsBinding struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Provider defines the provider type of the CredentialsBinding.
+	// This field is immutable.
+	Provider CredentialsBindingProvider
+	// CredentialsRef specify reference to credentials.
+	CredentialsRef Credentials
+	// Quotas is a list of references to Quota objects in the same or another namespace.
+	// This field is immutable.
+	Quotas []corev1.ObjectReference
+}
+
+// Credentials holds reference to credentials implementation.
+type Credentials struct {
+	// Secret is a reference to a secret object in the same or another namespace.
+	Secret *corev1.SecretReference
```

Comment:
I guess you didn't want to name this `SecretRef` to avoid cases where we would type `credentialsRef.secretRef`.
What about naming the `CredentialsRef` variable above to simply `Credentials` so that it corresponds to the structure and then naming `Secret` and `WorkloadIdentity` to `SecretRef` and `WorkloadIdentityRef`?

Reply:
I applied the proposed changed.